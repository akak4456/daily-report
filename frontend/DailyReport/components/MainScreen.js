import React, {Component} from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {Actions} from 'react-native-router-flux';

class MainScreen extends Component {
  constructor() {
    super();
    this.forceUpdateHandler = this.forceUpdateHandler.bind(this);
  }

  forceUpdateHandler() {
    this.forceUpdate();
  }

  render() {
    return (
      <View>
        <Text onPress={() => Actions.sub()}>메인asdflek</Text>
      </View>
    );
  }
}

export default MainScreen;
