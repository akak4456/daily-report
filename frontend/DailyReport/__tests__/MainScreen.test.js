import React from 'react';
import MainScreen from '../components/MainScreen';
import {render} from '@testing-library/react-native';

jest.mock('react-navigation', () => {});
jest.mock('react-native-gesture-handler', () => {});
jest.mock('react-navigation-stack', () => {});
jest.mock('react-native-reanimated', () => {});
jest.mock('react-navigation-drawer', () => {});
jest.mock('react-navigation-tabs', () => {});

describe('<MainScreen />', () => {
  test('renders without crashing', () => {
    const rendered = render(<MainScreen />);
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });
});
