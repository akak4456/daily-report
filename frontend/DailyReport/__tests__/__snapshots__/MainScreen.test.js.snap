// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainScreen /> renders without crashing 1`] = `
<View>
  <Text
    onPress={[Function]}
  >
    메인
  </Text>
</View>
`;
